### PR TITLE
Upgrade contracts to v0.4.0.alpha.1

### PR DESCRIPTION
**Pull request by alpe**
- Bump contracts
- More tests

### CODE REVIEW

**Comment by ethanfrey** [comment]
Great you started on this. Two notes (and please check with Mauro on slack to align e2e tests with contract design).

1. No ics20 channel is used. See https://github.com/osmosis-labs/mesh-security/blob/main/docs/ibc/Rewards.md#selected-implementation
2. Great to query that rewards show up on the provider side. When they are withdrawn, you must provide a consumer side address and the tokens will be released to that address (via the same control channel).

This is mean to handle accounting of the consumer rewards where the stakers are, but keep the rewards on their home chain, to encourage staking over selling

File: x/meshsecurity/keeper/scheduler.go [hunk 1]
```diff
@@ -33,6 +33,17 @@ func (k Keeper) HasScheduledTask(ctx sdk.Context, tp types.SchedulerTaskType, co
 	return err == nil && result // we can ignore the unknown task type error and return false instead
 }
 
+// GetNextScheduledTaskHeight returns height for task to execute
+func (k Keeper) GetNextScheduledTaskHeight(ctx sdk.Context, tp types.SchedulerTaskType, contract sdk.AccAddress) (height uint64, found bool) {
+	err := k.iterateScheduledContractTasks(ctx, tp, contract, math.MaxUint, func(atHeight uint64, _ bool) bool {
```
Comment (pinosu) [inline x/meshsecurity/keeper/scheduler.go:38]: 👍

File: demo/app/test_helpers.go [hunk 2]
```diff
@@ -168,14 +168,28 @@ func SetupWithGenesisValSet(t *testing.T, valSet *tmtypes.ValidatorSet, genAccs
...
-	}})
+
+	votes := make([]abci.VoteInfo, len(valSet.Validators))
+	for i, v := range valSet.Validators {
+		votes[i] = abci.VoteInfo{
+			Validator:       abci.Validator{Address: v.Address, Power: v.VotingPower},
+			SignedLastBlock: true,
+		}
+	}
+
```
Comment (pinosu) [inline demo/app/test_helpers.go:179]: 👍

File: tests/e2e/Makefile [hunk 3]
```diff
@@ -13,4 +13,10 @@ build:
 build-linux:
 	CGO_ENABLED=1  GOOS=linux go test -c -mod=readonly -o $(BUILD_DIR)/mesh-security.e2e.test
 
-.PHONY: all build build-linux test
+format:
+	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./x/vendor*" -not -path "./contracts*" -not -path "./packages*" -not -path "./docs*"| xargs misspell -w
+	find . -name '*.go' -type f -not -path "./vendor*" -not -path "*.git*" -not -path "./x/vendor*" -not -path "./contracts*" -not -path "./packages*" -not -path "./docs*"| xargs gofumpt -w -s
```
Comment (pinosu) [inline tests/e2e/Makefile:18]: 👍

**Review by pinosu — APPROVED** [review]
Great work! LGTM 👍

**Pull request merged**